src/services/polling.service.ts

```diff
@@ -0,0 +1,201 @@
+import { TransactionResponse, TransactionStatus } from "fireblocks-sdk";
+import { Clients } from "../interfaces/Clients";
+import { patchTransactionAmountUsd } from "../util/cmc/patchTransactionAmountUsd";
+import {
+  handleTransactionCreated,
+  handleTransactionCreatedOrUpdated,
+  handleTransactionStatusUpdated,
+} from "./webhook.service";
+import { Transaction } from "../model/transaction";
+import { MoreThan } from "typeorm";
+import {
+  ITransactionDetails,
+  TransactionSubStatus,
+} from "../interfaces/transaction";
+import { fetchAllTxs } from "../util/fetch-all";
+
+const sleep = (ms: number) => new Promise((resolve) => setTimeout(resolve, ms));
+
+const groupBy = <T, K extends keyof never>(arr: T[], key: (i: T) => K) =>
+  arr.reduce(
+    (groups, item) => {
+      (groups[key(item)] ||= []).push(item);
+      return groups;
+    },
+    {} as Record<K, T[]>,
+  );
+
+export class PollingService {
+  private active: boolean;
+  private readonly pollingIntervalMs: number = 60_000;
+  private readonly pollingShortIntervalMs: number = 10_000;
+  private currentPollingIntervalMs: number;
+  private incomingTxTimer: NodeJS.Timeout | null;
+
+  static readonly FINAL_STATUSES: TransactionStatus[] = [
+    TransactionStatus.COMPLETED,
+    TransactionStatus.CANCELLED,
+    TransactionStatus.REJECTED,
+    TransactionStatus.FAILED,
+    TransactionStatus.BLOCKED,
+  ];
+
+  constructor(private readonly clients: Clients) {
+    this.active = false;
+    this.currentPollingIntervalMs = this.pollingIntervalMs;
+  }
+
+  private async getDbRecentTransactionsSorted(
+    minLastUpdated: Date,
+  ): Promise<
+    Pick<Transaction, "id" | "createdAt" | "lastUpdated" | "status">[]
+  > {
+    return await Transaction.find({
+      select: ["id", "createdAt", "lastUpdated", "status"],
+      where: {
+        lastUpdated: MoreThan(minLastUpdated),
+      },
+      order: { createdAt: "ASC" },
+    });
+  }
+
+  private hoursAgoDate(hours: number) {
+    const date: Date = new Date();
+    date.setHours(date.getHours() - hours);
+    return date;
+  }
+
+  private isCompletedTx(tx: { status: TransactionStatus }) {
+    return PollingService.FINAL_STATUSES.includes(tx.status);
+  }
+
+  private txResponseToTxDetails(tr: TransactionResponse): ITransactionDetails {
+    return {
+      ...tr,
+      subStatus:
+        TransactionSubStatus[tr.subStatus as keyof typeof TransactionSubStatus],
+      destinations: tr.destinations?.map((dest) => ({
+        ...dest,
+        amountUSD: Number(dest.amountUSD),
+      })),
+      signedMessages: tr.signedMessages!,
+    };
+  }
+
+  private async pollAndUpdateAll() {
+    console.log("PollingService: pollAndUpdateAll");
+    try {
+      // Get transactions from Fireblocks:
+      const txsResponses: TransactionResponse[] = await fetchAllTxs(
```

Comment:
fetching all txs can be pretty heavy, and not all txs are required.
i'd consider:
1. fetching only txs from\to END_USER_WALLETS
2. like 1 but also for a single NCW at a time, per request